### ISSUE TITLE
Travis build failing for approximation/kcomponents

### ISSUE THREAD

**Issue by MridulS** [networkx/networkx#1637]
https://travis-ci.org/networkx/networkx/jobs/69143172#L3006
https://travis-ci.org/networkx/networkx/jobs/67124137#L3417
https://travis-ci.org/networkx/networkx/jobs/66950067#L2992

EDIT: These links are redundant now because they have been restarted
The output for the failed test.
``` python
FAIL: test_kcomponents.test_configuration
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/networkx/algorithms/approximation/tests/test_kcomponents.py", line 148, in test_configuration
    _check_connectivity(G)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/networkx/algorithms/approximation/tests/test_kcomponents.py", line 125, in _check_connectivity
    assert_true(K >= k)
nose.proxy.AssertionError: False is not true
    'False is not true' = self._formatMessage('False is not true', "%s is not true" % safe_repr(False))
>>  raise self.failureException('False is not true')
```

**Comment by jtorrents** [comment]
Hmmm, I cannot reproduce this locally. I restarted the failing build and now it passes. When I was working on the `k_components` this specific test (`test_configuration`) failed at least once in Travis (I was never able to reproduce in my machine, and at Travis passes most of the time).

Any idea on how to debug this further?  Maybe the best approach is to just remove this test.

**Comment by MridulS** [comment]
Even I couldn't reproduce this failure. Yes I think we should remove this test for the time being but this failure has appeared multiple times, there must be something wrong going on in there.

**Comment by MridulS** [comment]
Did you restart the build for https://travis-ci.org/networkx/networkx/jobs/67124137#L3417 ?

**Comment by jtorrents** [comment]
Not before, but I just did.

**Comment by jtorrents** [comment]
I'm testing with different values of `PYTHONHASHSEED` to make sure that this error is not due to iteration order. No failures so far. I'm testing it using this approach:

```bash
for i in {100000..1000000..50000}; do export PYTHONHASHSEED=$i; echo PYTHONHASHSEED=$i; PYTHONHASHSEED=$i nosetests3 -v networkx/algorithms/approximation/tests/test_kcomponents.py; done
``` 

I'm trying different ranges, but the number of possible hash seeds is huge.

**Comment by jtorrents** [comment]
Fixed in #1639 and #1658
